applications/OptimizationApplication/python_scripts/processes/optimization_problem_vtu_output_process.py

```diff
@@ -151,16 +167,27 @@ def __init__(self, parameters: Kratos.Parameters, optimization_problem: Optimiza
             raise RuntimeError(f"Only supports \"ascii\" and \"binary\" file_format. [ provided file_format = \"{file_format}\" ].")
 
         self.list_of_components: 'list[Union[str, ResponseFunction, Control, ExecutionPolicy]]' = []
-        list_of_component_names = parameters["list_of_output_components"].GetStringArray()
-        if len(list_of_component_names) == 1 and list_of_component_names[0] == "all":
-            list_of_component_names = GetAllComponentFullNamesWithData(optimization_problem)
-
-        for component_name in list_of_component_names:
-            self.list_of_components.append(GetComponentHavingDataByFullName(component_name, optimization_problem))
-
+        self.list_of_component_names = parameters["list_of_output_components"].GetStringArray()
         self.list_of_expresson_vtu_outputs: 'list[ExpressionVtuOutput]' = []
         self.initialized_vtu_outputs = False
 
+    def ExecuteInitialize(self) -> None:
+        # get all the component names at the first writing point
+        if len(self.list_of_component_names) == 1 and self.list_of_component_names[0] == "all":
+            self.list_of_component_names = GetAllComponentFullNamesWithData(self.optimization_problem)
+
+        for component_name in self.list_of_component_names:
+            self.list_of_components.append(GetComponentHavingDataByFullName(component_name, self.optimization_problem))
+
+        optimization_problem_data = self.optimization_problem.GetProblemDataContainer()
+
+        # check if there are vtu output requests from some other processes
+        if not optimization_problem_data.HasValue("requested_container_expression_outputs"):
+            optimization_problem_data["requested_container_expression_outputs"] = set()
+
+        for component in self.list_of_components:
+            optimization_problem_data["requested_container_expression_outputs"].add(component)
```

Comment:
This thing is not required anymore, and to be honest, this creates a circular dependency which is not so good.